packages/@webex/plugin-presence/src/presence.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) 2015-2020 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import '@webex/internal-plugin-device';
+
+import {WebexPlugin} from '@webex/webex-core';
+
+import PresenceBatcher from './presence-batcher';
+import PresenceWorker from './presence-worker';
+import {IEventPayload, IPresence, IPresenceStatusObject} from './interface';
+
+const defaultSubscriptionTtl = 600;
+const USER = 'user';
+const USER_PRESENCE_ENABLED = 'user-presence-enabled';
+
+/**
+ * @class
+ * @extends WebexPlugin
+ */
+const Presence: IPresence = WebexPlugin.extend({
+  namespace: 'Presence',
+
+  children: {
+    batcher: PresenceBatcher,
+  },
+
+  session: {
+    worker: {
+      default() {
+        return new PresenceWorker();
+      },
+      type: 'any',
+    },
+  },
+
+  /**
+   * Initialize the presence worker for client
+   * @returns {undefined}
+   */
+  initialize(): void {
+    this.webex.once('ready', () => {
+      if (this.config.initializeWorker) {
+        this.worker.initialize(this.webex);
+      }
+    });
+  },
+
+  /**
+   * Trigger an event.
+   * @param {string} event
+   * @param {string} payload
+   * @returns {undefined}
+   */
+  emitEvent(event: string, payload: IEventPayload): void {
+    if (payload.type && payload.payload) {
+      this.trigger(event, payload);
+    }
+  },
+
+  /**
+   * Enables presence feature
+   * @returns {Promise<boolean>} resolves with true, if successful
+   */
+  enable(): Promise<boolean> {
+    return this.webex.internal.feature
+      .setFeature(USER, USER_PRESENCE_ENABLED, true)
+      .then((response) => response.value);
+  },
+
+  /**
+   * Disables presence feature
+   * @returns {Promise<boolean>} resolves with false, if successful
+   */
+  disable(): Promise<boolean> {
```

Comment:
im getting a bunch of suggestions to change these to async functions
![image](https://github.com/webex/webex-js-sdk/assets/14194394/0974a38a-ff68-4dd6-b85d-a98df83837fd)

Reply:
Correct, but I haven't changed the code from the old `thenable` syntax to async-await so there is no need to convert them to async function.

In this PR, the goal is to make the package public and add types. Maybe we can refactor this in the future if we start to get changes in functionality, for now, it should be fine.